File: medplum-java/medplum-java-model/src/main/java/com/fhir/Patient_Contact.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutablePatient_Contact.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Patient_Contact {
  Optional<String> id();

  Optional<Period> period();

  Optional<List<Extension>> modifierExtension();

  Optional<Reference> organization();

  Optional<Address> address();

  Optional<Patient_contactGender> gender();

  Optional<HumanName> name();

  Optional<List<CodeableConcept>> relationship();

  Optional<List<ContactPoint>> telecom();

  Optional<List<Extension>> extension();

  static ImmutablePatient_Contact.Builder builder() {
    return ImmutablePatient_Contact.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/Patient_Contact.ts
import { Address } from "./Address";
import { CodeableConcept } from "./CodeableConcept";
import { ContactPoint } from "./ContactPoint";
import { Extension } from "./Extension";
import { HumanName } from "./HumanName";
import { Patient_contactGender } from "./Patient_contactGender";
import { Period } from "./Period";
import { Reference } from "./Reference";

export interface Patient_Contact {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    relationship: CodeableConcept[] | null | undefined;
    name: HumanName | null | undefined;
    telecom: ContactPoint[] | null | undefined;
    address: Address | null | undefined;
    gender: Patient_contactGender | null | undefined;
    organization: Reference | null | undefined;
    period: Period | null | undefined;
}
